app/graphql/mutations/sign_in_musician.rb

```diff
@@ -14,16 +14,16 @@ def resolve(credentials: nil)
       musician = Musician.find_by email: credentials[:email]
 
       # ensures we have the correct user
-      return unless musician
-      return unless musician.authenticate(credentials[:password])
-
-      len = ActiveSupport::MessageEncryptor.key_len
-      salt  = SecureRandom.random_bytes(len)
-      key   = ActiveSupport::KeyGenerator.new('password').generate_key(salt, len)
-      crypt = ActiveSupport::MessageEncryptor.new(key)
-      token = crypt.encrypt_and_sign("musician-id:#{ musician.id }")
-
-      { musician: musician, token: token }
+      if musician && musician.authenticate(credentials[:password])
+        len = ActiveSupport::MessageEncryptor.key_len
```

Comment:
all these variables looks really clutter and hard to read in the `if`. What do you think about moving them above the `if` or into an `initialize`?

Reply:
Good point! Moved them outside the if block.